buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/versioning/BuildVersionPlugin.kt

```diff
@@ -33,16 +33,27 @@ import org.gradle.gradlebuild.BuildEnvironment.CI_ENVIRONMENT_VARIABLE
 import org.gradle.kotlin.dsl.*
 import java.text.SimpleDateFormat
 import java.util.Date
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import ext
 
 
 class BuildVersionPlugin : Plugin<Project> {
     override fun apply(project: Project) {
         require(project === project.rootProject)
         project.setBuildVersion()
+        project.setBuildGitVersion()
     }
 }
 
 
+private
+fun Project.setBuildGitVersion() {
+    val repository = FileRepositoryBuilder().setGitDir(rootProject.projectDir.resolve(".git")).build()
+    ext["gradleBuildBranch"] = System.getenv("BUILD_BRANCH") ?: repository.branch
```

Comment:
:o: It would be really nice if we could stop using "ext", which is something discouraged, in our own build. This is also important for instant execution. Could we introduce a `BuildVersionExtension` here instead to carry the two values? 

Also, it would be better to use `providers.environmentVariable("BUILD_BRANCH")` instead of `System.getenv("BUILD_BRANCH") `. Not sure if it matters here, but to be on the save side and to be consistent (we should use that everywhere in our build). The `BuildVersionExtension` should then then define `Property<String>` properties so we can pass in the provider and resolve the value when it is needed.

Reply:
I used `GitInformationExtension` instead. Quite a few tasks are using this information so it will be a little trouble if it's a `Provider`.